doc/user_guide/psyir.rst

```diff
@@ -122,15 +122,43 @@ The ``SymbolTable`` has the following interface:
 .. autoclass:: psyclone.psyir.symbols.SymbolTable
     :members:
 
-Where each element is a `Symbol` with an immutable name:
+Where each element is a ``Symbol`` with an immutable name:
 
 .. autoclass:: psyclone.psyir.symbols.Symbol
     :members:
 
-There are several `Symbol` sub-classes to represent different labeled entities
+There are several ``Symbol`` sub-classes to represent different labeled entities
 in the PSyIR. At the moment the available symbols are:
 
 - .. autoclass:: psyclone.psyir.symbols.ContainerSymbol
 
 - .. autoclass:: psyclone.psyir.symbols.DataSymbol
 
+As described in the ``DataSymbol`` constructor, the ``shape``
+specification not only describes whether or not the ``Symbol`` is an
+array but also, if it is an array, captures what is known about how the extent
+(size) of each dimension is managed. This is necessary to distinguish
+between four cases:
+
+.. tabularcolumns:: |p{9cm}|L|
```

Comment:
Not part of this PR, but the table does not break into paragraphs and uses long lines instead (at least in html).

Also, do the description sentences need punctuation?

Reply:
Yes, I'm happy to leave long lines in html (although I know Joerg doesn't like it). I've added punctuation.